scripts/convert-svgs-to-icon-mixins.js

```diff
@@ -8,39 +8,109 @@ const directory = args[2];
 const symbols = /[\r\n%#()<>?\[\\\]^`{|}]/g;
 const svgo = new SVGO();
 
-function encodeSVG(data) {
+function parseColorArguments(data) {
+  const color_count = (data.match(/fill="#[^\s]+"/g) || []).length;
+  let arguments = [];
+
+  if (color_count == 1) {
+    arguments.push('$color');
+  } else {
+    for (var i = 0; i < color_count; i++) {
+      arguments.push(`$color${i + 1}`);
+    }
+  }
+
+  return arguments;
+}
+
+function printBaseMixins(svgs) {
+  console.log('// Base definition mixins:');
+  console.log('//------------------------');
+
+  svgs.forEach((svg) => {
+    console.log(`@mixin vf-icon-${svg.name}(${svg.colors.join(', ')}) {`);
+    console.log(` background-image: url("data:image/svg+xml,${encodeSVG(svg.svg, svg.colors)}");`);
+    console.log('}');
+    console.log('');
+  });
+}
+
+function printPatternMixins(svgs) {
+  console.log('// Pattern includes:');
+  console.log('//------------------');
+
+  svgs.forEach((svg) => {
+    console.log(`@include vf-p-icon-${svg.name};`);
+  });
+
+  console.log('// Pattern mixins:');
+  console.log('//----------------');
+
+  svgs.forEach((svg) => {
```

Comment:
I'm a bit concerned about adding these pattern mixins here. I understand their usefulness in terms of just being able to copy them into the source, but they are very tied to the structure of SCSS code in Vanilla.
The base mixins are also Vanilla code, but practically it's just a `background-image` definition in a mixin. Nothing more. Pattern mixins need to be exactly what we have in pattern code in Vanilla, together with extends, correct mixin names, nested selectors...

If we happen to change anything in the icons code we would need to remember to change the script as well.

I don't know what's the best way to approach it. Maybe adding comments both to icons file, mentioning that these were generated by the script and to the script to warn the user in the future to verify if the printed code is correct.

Reply:
I originally had just the `background-image` definition, and began adding them that way before quickly finding that adding all of these icons meant a lot of repetitive naming of mixins, and creating pattern mixins. Even with just a handful it's time consuming, so I think adding a disclaimer to the printed code is a reasonable solution.